docs/DeveloperGuide.md

```diff
@@ -1,29 +1,115 @@
+[DeveloperGuide.md](DeveloperGuide.md)
 # Developer Guide
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Command Mechanisms:
+For all valid commands, the mechanism of implementation are as follows:
+1. Read input - ```runBagPacker()``` method in ```BagPacker``` calls the ```Parser``` class to read user input command
+2. Create command object - The ```Parser``` class creates a corresponding command object of the relevant command
+3. Execute command object - ```runBagPacker()``` method executes the ```.execute()``` method (overridden by child classes) of the command object 
+   which runs the actual command function
+
+#### Add Command
+
+Add command is used to add a quantity of item(s) to the packing list.
+
+Mechanism: ```AddCommand.execute()``` calls the ```PackingList.addItem()``` method from the ```PackingList``` class which executes the ```ArrayList.add()``` method to add the item to the ```PackingList``` ArrayList. 
+It then updates the ```quantity``` variable according to the quantity inputted by the user.
+
+
+#### Preventing duplicate items
```

Comment:
`Preventing duplicate items` and `add command` now are on the same layer which makes it ambiguous. Since `preventing duplicate items` is under `add command` then it should be one layer down